src/Engine/ProtoCore/ProcedureTable.cs

```diff
@@ -396,5 +372,37 @@ public int IndexOf(string name, List<ProtoCore.Type> argTypeList, bool isStaticO
             }
             return currentProcedure;
         }
+
+        /// <summary>
+        /// Get function by its signature.
+        /// </summary>
+        /// <param name="functionName"></param>
+        /// <param name="parameterTypes"></param>
+        /// <returns></returns>
+        public ProcedureNode GetFunctionBySignature(string functionName, List<Type> parameterTypes)
+        {
+            foreach (var f in Procedures)
+            {
+                if (f.Name != functionName || f.ArgumentTypes.Count != parameterTypes.Count)
+                {
+                    continue;
+                }
+                
+                for (int i = 0; i < parameterTypes.Count; i++)
+                {
+                    if (f.ArgumentTypes[i].Name != parameterTypes[i].Name || f.ArgumentTypes[i].UID != parameterTypes[i].UID)
+                    {
+                        goto NotMatch; 
+                    }
+                }
+
+                return f;
```

Comment:
Where do you check for the rank of the parameters here in order to return the function with the highest ranks?

Reply:
First come first serve